openedx/core/djangoapps/discussions/models.py

```diff
@@ -2,19 +2,124 @@
+from organizations.models import Organization
 
+from openedx.core.djangoapps.config_model_utils.models import StackedConfigurationModel
 from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
 
+log = logging.getLogger(__name__)
+
+
+def get_default_providers():
+    providers = [
+        'cs_comments_service',
+        'lti-base',
+    ]
+    return providers
+
+
+class ProviderFilter(StackedConfigurationModel):
+    allow = ListCharField(
+        base_field=models.CharField(
+            choices=[
+                (provider, provider)
+                for provider in get_default_providers()
+            ],
+            max_length=20,
+        ),
+        blank=True,
+        help_text=_("Comma-separated list of providers to allow, eg: {choices}").format(
+            choices=','.join(get_default_providers()),
+        ),
+        # max_length = (size * (max_length + 1)) # to include the comma
+        max_length=(len(get_default_providers()) * 21),
+        size=len(get_default_providers()),
+        verbose_name=_('Allow List'),
+    )
+    deny = ListCharField(
+        base_field=models.CharField(
+            choices=[
+                (provider, provider)
+                for provider in get_default_providers()
+            ],
+            max_length=20,
+        ),
+        blank=True,
+        help_text=_("Comma-separated list of providers to deny, eg: {choices}").format(
+            choices=','.join(get_default_providers()),
+        ),
+        # max_length = (size * (max_length + 1)) # to include the comma
+        max_length=(len(get_default_providers()) * 21),
+        size=len(get_default_providers()),
+        verbose_name=_('Deny List'),
+    )
+
+    STACKABLE_FIELDS = ('allow', 'deny')
+
+    @classmethod
+    def default_providers(cls) -> list[str]:
+        # TODO: Load this from the entry points
+        return []
+
+    @property
+    def allow_list(self) -> list[str]:
+        if self.allow:
+            return self.allow
+        return []
+
+    @property
+    def deny_list(self) -> list[str]:
+        if self.deny:
+            return self.deny
+        return []
+
+    def __str__(self):
+        return 'ProviderFilter(org="{org}", allow="{allow}", deny="{deny}")'.format(
+            allow=self.allow,
+            deny=self.deny,
+            org=self.org,
+        )
+
+    @property
+    def available_providers(self) -> list[str]:
+        _providers = ProviderFilter.current().default_providers()
+        if self.allow_list:
+            _providers = [
+                provider
+                for provider in _providers
+                if provider in self.allow_list
+            ]
+        if self.deny_list:
+            _providers = [
+                provider
+                for provider in _providers
+                if provider not in self.deny_list
+            ]
+        return _providers
+
+    @classmethod
+    def get(cls, course_key) -> cls:
```

Comment:
I'm not sure if this helper is necessary and/or adds anything...
Maybe just remove it and inline the invocation;
it's just `cls.current(course_key=course_key)`